sympy/polys/rings.py

```diff
@@ -1627,9 +1672,17 @@ def evaluate(self, *args, **kwargs):
         else:
             temp_result = self._subs(eval_dict)
             new_ring = ring.drop(*[ring.gens[i] for i in eval_dict.keys()])
-            return temp_result.set_ring(new_ring)
+            return temp_result.set_ring(new_ring) # type: ignore
 
-    def subs(self, *args, **kwargs):
+    @overload
+    def subs(self, values: list[tuple[Expr, Er | int]]) -> PolyElement[Er]:
+        ...
+
+    @overload
+    def subs(self, x: Expr, a: Er | int) -> PolyElement[Er]:
+        ...
```

Comment:
Expr is not correct here. The value gets passed to `ring.index` which expects `PolyElement[Er] | int | str`.